### ISSUE TITLE
Screen shake and dash intensity.

### ISSUE THREAD

**Issue by Shodance** [Moffein/HAN-D_OVERCLOCKED#5]
Not a game breaking issue by any means but the screen shake from han-d's second Primary and han-d's Secondary are really intense.

Also the dash from the second Primary is really disorienting, I think it would be better if it was removed?

**Comment by Moffein** [comment]
I can adjust the shake. Could you go into more detail about what feels disorienting about the alt primary's lunge?

**Comment by Shodance** [comment]
I feel the lunge and camera shake itself its the problem, the combination of the very quick, short dash and the large camera shake independent of hitting an enemy is the issue.

Comparing the camera shake to other chars like Enforcer/Nemforcer its very harsh.

You are also unable to strafe around enemies while attacking while using the alt primary, doubly so since it doesn't take directional input into account, you dash directly forward at a fixed distance.

I feel the lunge aspect also makes it disorienting to fight enemies up close, especially bosses like the titans, you will need to keep swivelling your view 180 degrees because you end up behind them, even when holding backwards.

I played a lot of the original HAN-D mod so I feel very strongly about this kind of stuff. Over all this new version is great tho.

Maybe an alternative could be removing the speed debuff during the hit portion of the attack, instead of the dash, imo.

**Comment by Moffein** [comment]
Noted, will try to make some tweaks. I like the mechanical feel of the lunge so I'd like to keep it in some form.
Some tweaks I have in mind are:
- Slow down the base lunge speed a bit (which in turn will lower the distance)
- Each enemy hit slows down the lunge further (Scale off of enemy mass, so you stop at boss enemies? Could be jank, need to test.)
- Movement direction influences lunge trajectory
    - In the current version, not pressing any movement keys disables the lunge.
    - Could make it so that lunge speed scales from 100% -> 50% based on how far away your movement direction is from your aim direction.

**Comment by Shodance** [comment]
Those all sound like great changes :D

**Comment by Moffein** [comment]
[HAND_Overclocked.zip](https://github.com/Moffein/HAN-D_OVERCLOCKED/files/10127668/HAND_Overclocked.zip)
Could you test this out and let me know how it feels? Changes to the Hammer M1 in this version are:
- Reduced screenshake amplitude from 5 -> 3.9
- Now only lunges forward if the forward movement button is pressed.
	- On gamepad, checks if movement input is within 50 degrees of the forward direction on either side.
- Each enemy hit reduces lunge speed by 50%.

**Comment by Shodance** [comment]
Big improvements on the lunge, much more control over the direction I want at any given time, I like that a lot.

Vid example against a titan, strafing around, dashing forward when needed, very fun to control; https://www.youtube.com/watch?v=tqgGfzfD7RI

The camera shake reduction is definitely noticeable but I still feel its a bit too high. However, there's something else I noticed when comparing it to other forms of camera shake;

I tested Enforcer, Nemforcer, and Acrid, and all of them have camera shake that pretty closely matches their actions.

Enforcers shield bash makes the camera shake diagonally up left, the same direction that he bashes his shield. The recoil from his primary shakes the camera vertically.

Nemforcers smash attack does a harsh vertical shake, and his primary gun shakes mostly vertical with slight pivots left and right to match the gatling guns radius. In fact, I wish his primary axe swing had something Acrid has:

Acrids three hit swipe seems to shake the camera right, left, and then vertically, to match the three hits of his swipe.

Vid example comparing shake: https://www.youtube.com/watch?v=z1Tp-WIGxBA

If HAN-D's left and right hammer swings could have camera shake that primarily pivots left, right, left, right instead of all around, I think that would go a long way to reduce on motion sickness, and improve game feel!

Again, that's just me, and I hope I don't sound like a crack pot saying all this.
